cmd/sd/internal/api/auth.go

```
@@ -0,0 +1,269 @@
+	cfg.UpsertContext(s.contextName, currentContext)
+
+	return s.store.Save(cfg)
+}
+
+func refreshError(contextName string, apiURL string, token *openapi.OAuthTokenResponse) error {
+	if token.JSON400 != nil {
+		switch token.JSON400.Error {
+		case "invalid_grant":
+			return reauthenticateError("context %q session expired or was revoked", contextName)
+		case "invalid_client":
+			return fmt.Errorf("Storyden CLI OAuth client was rejected by %s", apiURL)
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not be capitalized

(staticcheck)